packages/runner/src/collect.ts

```diff
@@ -41,24 +43,26 @@ export async function collectTests(paths: string[], runner: VitestRunner): Promi
 
       const defaultTasks = await getDefaultSuite().collect(file)
 
-      setHooks(file, getHooks(defaultTasks))
+      const fileHooks = createSuiteHooks()
+      mergeHooks(fileHooks, getHooks(defaultTasks))
 
       for (const c of [...defaultTasks.tasks, ...collectorContext.tasks]) {
-        if (c.type === 'test') {
-          file.tasks.push(c)
-        }
-        else if (c.type === 'custom') {
-          file.tasks.push(c)
-        }
-        else if (c.type === 'suite') {
+        if (c.type === 'test' || c.type === 'custom' || c.type === 'suite') {
           file.tasks.push(c)
         }
         else if (c.type === 'collector') {
           const suite = await c.collect(file)
-          if (suite.name || suite.tasks.length)
+          if (suite.name || suite.tasks.length) {
+            mergeHooks(fileHooks, getHooks(suite))
             file.tasks.push(suite)
+          }
+        }
+        else {
+          c satisfies never
```

Comment:
Is this intentional? What's it for?

Reply:
Make sure all options are exhausted - if we add another type, this will throw a type error

Reply:
Oh right, that's nice trick. 👍 

It might be worth to add describing comment above this. Otherwise it looks like this line isn't really doing anything.